lib/utils/job-utils/racadm-tool.js

```diff
@@ -192,29 +220,136 @@ function racadmFactory(
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @param {Object} cifsConfig -  includes samba password, username and bios.xml file path
+     * Password and username are only required when fiel path is a cifs share folder.
+     * @param {string} device - Dell device Full-Qualified-Device-Descriptor (FQDD)
+     * @returns {object} - job status
+     */
+    RacadmTool.prototype.getBiosConfig = function(host, user, password, cifsConfig, device) {
+        var self =  this;
+        var FILEPATH = '/tmp/configure.xml'; //root authority required
+        var command = '', fqdd, cifsUser, cifsPassword, fileStoredPath;
+        cifsConfig = cifsConfig || {};
+        cifsUser = cifsConfig.user || '';
+        cifsPassword = cifsConfig.password || '';
+        fileStoredPath = cifsConfig.filePath || FILEPATH;
+
+        return Promise.resolve()
+            .then(function(){
+                return self.getSoftwareList(host, user, password);
+            })
+            .then(function(softwareList){
+                if (typeof softwareList.BIOS.FQDD === "undefined"){
+                    throw new Error ('Can not get BIOS FQDD');
+                }
+                fqdd = device || softwareList.BIOS.FQDD;
+            })
+            .then(function(){
+                return parser.getPathFilename(fileStoredPath);
+            })
+            .then(function(fileInfo){
+                //wrap commands
+                if (fileInfo.style === 'remote') {
+                    command = "get -f " + fileInfo.name + " -t xml -u " + cifsUser +
+                        " -p " + cifsPassword + " -l " + fileInfo.path + " -c " + fqdd;
+                } else{ // 'local'
+                    command = "get -f " + fileStoredPath + " -t xml -c " + fqdd;
+                }
+
+                return command;
+            }).then(function(command){
+                return self.runAsyncCommands(host, user, password, command, 0, 1000);
+            });
+    };
 
-        return this.runAsynCommands(host, user, password, command, 0, 1000);
+    /**
+     * Get all configure info from iDRAC
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @returns {object} - all iDRAC configure info in jsonn format
+     */
+    RacadmTool.prototype.getConfigCatalog = function(host, user, password) {
+        var filePath = '/tmp/idrac_configure.xml', self = this,
+            command =  'get -f ' + filePath + ' -t xml';
+        return Promise.resolve()
+            .then(function(){
+                return self.runCommand(host, user, password, command, 0, 1000);
+            })
+            .then(function(){
+                fs.exists(filePath, function(exist){
+                    if(!exist) {
+                        throw new Error('File /tmp/idra_configure.xml does not exist');
+                    }
```

Comment:
as `fs.exists` is an async function, so the error will not be threw as your design. You need to warp this function to Promise if you want to use it in the promise chain.